auth/auth.go

```diff
@@ -100,10 +100,11 @@ type Token struct {
 	// response returned by the server.
 	Metadata map[string]interface{} // TODO(codyoss): maybe make a method to flatten metadata to avoid []string for url.Values
 	// TrustBoundaryData represents the trust boundary data associated with the token.
-	// It contains information about the regions or environments where the token is valid.
-	// This data is used to enforce trust boundary restrictions on requests made with the token.
-	// TrustBoundaryData is a value type for immutability after token creation.
-	TrustBoundaryData internal.TrustBoundaryData
+	// It contains information about the regions or environments where the token is
+	// valid. This data is used to enforce trust boundary restrictions on requests
+	// made with the token. trustBoundaryData is a value type for immutability
+	// after token creation.
+	trustBoundaryData internal.TrustBoundaryData
```

Comment:
I overall like this approach much better, thank you for the refactoring. I think we can take it one step further and limit the new public fields/methods to just TokenHook. I think we should make use of the Metadata field here to store this data. We do this already today for some compute usecases, please see compute.go for an example. We should create a well-defined key that we can use to store and retrieve the values required for this validation.

Reply:
Sounds good. I've moved the trust boundary data to the metadata field. Let me know what you think.